vignettes/betareg.Rmd

```diff
@@ -0,0 +1,195 @@
+dimensional vector of parameters associated with the $n\times L$ dimensional predictor matrix $Z$. 
+
+# Priors
+
+```{r, child="children/stan_glm_priors.txt"}
+```
+
+# Posterior
+
+When using only a *single set of regressors*, the posterior distribution of $\eta$ and $\phi$ is
+proportional to the product of the likelihood contributions, priors on the $K$ $\eta$ parameters, and $\phi$,
+$$
+f(\boldsymbol{\eta},\phi|\mathbf{y},\mathbf{X}) \propto 
+\prod_{i=1}^n f(y_i | \alpha,\beta) \times 
+\prod_{i=1}^K f(\boldsymbol{\eta}) \times
+f(\phi)
+$$
+
+When using *two sets of regressors*, the posterior distribution of $\eta$ and $\gamma$ is
+proportional to the product of the likelihood contribution, the $K$ priors on the $\eta$ parameters, and the
+$L$ priors on the $\gamma$ parameters,
+
+$$
+f(\boldsymbol{\eta},\boldsymbol{\gamma}|\mathbf{y},\mathbf{X}) \propto 
+\prod_{i=1}^n f(y_i | \alpha,\beta) \times 
+\prod_{k=1}^K f(\boldsymbol{\eta}) \times
+\prod_{l=1}^L f(\boldsymbol{\gamma})
+$$
+
+# An Example Using Simulated Data
+
+In this example the outcome variable $\mathbf{y}$ is simulated in a way that warrants the use of beta
+regression. It is worth mentioning that the data generation process is quite convoluted,
+which is apparent in the identification of the likelihood above. 
+
+The data simulated below uses the logistic link function on the first set of regressors
+and the log link function on the second set of regressors. 
+
+```{r simulated-data, fig.height=5}
+dat <- list()
+dat$eta <- c(1, -0.2)
+dat$gamma <- c(1.8, 0.4)
+dat$N <- 500
+dat$x <- rnorm(dat$N, 2, 2)
+dat$z <- rnorm(dat$N, 2, 2)
+dat$mu <- binomial(link = logit)$linkinv(dat$eta[1] + dat$eta[2]*dat$x)
+dat$phi <- binomial(link = log)$linkinv(dat$gamma[1] + dat$gamma[2]*dat$z)
+dat$y <- rbeta(dat$N, dat$mu * dat$phi, (1 - dat$mu) * dat$phi)
+dat <- data.frame(cbind(dat$y, dat$x, dat$z))
+colnames(dat) <- c("y", "x", "z")
+hist(dat$y, col = "darkgrey", border = F, main = "Distribution of Outcome Variable", xlab = "y", breaks = 20, freq = F)
+```
+
+The model can be fit by calling `stan_betareg`, using the appropriate link functions.
+```{r simulated-fit, results = "hide"}
+library(rstanarm)
+fit1 <- stan_betareg(y ~ x | z, data = dat, link = "logit", link.phi = "log", algorithm = "sampling")
+fit2 <- stan_betareg(y ~ x, data = dat, link = "logit", link.phi = "log", algorithm = "sampling")
+round(coef(fit1), 2)
+round(coef(fit2), 2)
+```
+
+``` {r simulated-fit-print, echo=FALSE}
+round(coef(fit1), 2)
+round(coef(fit2), 2)
+```
+
+The usual posterior analyses are available in **rstanarm**.
+The plots below illustrate simulated values of the outcome variable. The incorrect
+model noticeably fails to capture the top of the distribution consistently in comparison
+to the true model.
+```{r simulated-analysis, fig.height=5}
+library(gridExtra)
+plot1 <- pp_check(fit1) + ggtitle("True Model: y ~ x | z")
+plot2 <- pp_check(fit2) + ggtitle("False Model: y ~ x")
```

Comment:
Are these still not looking right? If not then I need to spend some time figuring out what's up with that.